cmd/flux/logs.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2021 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/fluxcd/flux2/internal/utils"
+	"github.com/spf13/cobra"
+	"html/template"
+	"io"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"os"
+	"strings"
+	"sync"
+)
+
+var logsCmd = &cobra.Command{
+	Use:   "logs",
+	Short: "Display formatted logs for toolkit components",
+	Long:  "",
+	RunE:  logsCmdRun,
+}
+
+type logsFlags struct {
+	level     string
+	watch     bool
+	tail      int64
+	kind      string
+	name      string
+	namespace string
+}
+
+var logsArgs = &logsFlags{
+	tail: -1,
+}
+
+func init() {
+	logsCmd.PersistentFlags().StringVarP(&logsArgs.level, "level", "l", logsArgs.level, "Level of logs to be displayed e.g error")
+	logsCmd.PersistentFlags().StringVarP(&logsArgs.kind, "kind", "", logsArgs.kind, "displays errors of a particular kind in flux")
+	logsCmd.PersistentFlags().StringVarP(&logsArgs.name, "name", "", logsArgs.name, "specifies the name of the object logs to be displayed")
+	logsCmd.PersistentFlags().BoolVarP(&logsArgs.watch, "watch", "w", logsArgs.watch, "Specifies if the logs should be streamed")
+	logsCmd.PersistentFlags().Int64VarP(&logsArgs.tail, "tail", "", logsArgs.tail, "Lines of recent log file to display")
+	rootCmd.AddCommand(logsCmd)
+}
+
+func logsCmdRun(cmd *cobra.Command, args []string) error {
+	const fluxSelector = "app.kubernetes.io/instance=flux-system"
```

Comment:
This can’t be a constant, the selector value should be the namespace from root args.